BrainPortal/lib/boutiques_input_cache_cleaner.rb

```diff
@@ -0,0 +1,78 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2021
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# This module adds automatic verification of the
+# type of files selected in a File input of a Boutiques Task.
+#
+# To include the module automatically at boot time
+# in a task integrated by Boutiques, add a new entry
+# in the 'custom' section of the descriptor, like this:
+#
+#   "custom": {
+#       "cbrain:integrator_modules": {
+#           "BoutiquesInputCacheCleaner": [
+#             "my_input1",
+#             "my_input2"
+#           ]
+#       }
+#   }
+#
+# In the example above, any userfile cache selected for the file input
+# named 'my_input1' or 'my_input2' will be deleted after task execution unless CBRAIN
+# notices that another task uses same cache.
+# CBRAIN tries to handle conflict between tasks that share same file based on timestamps.
+# Yet this is a dangerous feature as it might not handle some of races, e.g. with legacy tools
+module BoutiquesInputCacheCleaner
+
+  def setup #:nodoc:
+    result = super
+    self.meta[:setup_time] = Time.now
+    result
+  end
+
+  def save_results #:nodoc:
+
+    return fail unless super # call all the normal code
+
+    setup_time = self.meta[:setup_time] || Time.now
```

Comment:
That's wrong, if you don't have a setup time, just don't do anything and return instead. There is nothing to do without it!